woocommerce/woocommerce-android
# Issue/11674 stock card missing parts

Closes: #11674 


### Description

Adds a miscellanea of missing feature to the product stock card: 
- In memory cache to avoid unnecessary requests to fetch sales reports when stock report hasn't changed
- Make `Stock` card unavailable when there are no published products
- Handle empty case for `Stock` card, when API return empty report. 

### Testing information
1. Log into a site that doesn't have any products published
2. Check the `Stock` card is unavailable
3. Publish 1 product and don't change the default stock option.
4. Check the empty case is displayed.
5. Add a couple products with `low stock` status. Essentially set product stock to 1  for one product and 2 for the other product. 
6. Check the `Stock` card is now available for selection
7. Add `Stock` card to the dashboard and check the report is fetched correctly
8. Remove all cards from the dashboard except for the `Stock` card. 
9. Launch flipper (or any request interceptor tool) and verify that whenever we pull to refresh, the only request fired is `
/wc-analytics/reports/stock/` 
10. Make an order for the product with 2 available units.
11. Verify that the `Stock` card is refreshed and correctly reflects the new sales.
